elastic/apm-server
# Set `<event>.duration.us` field in ingest pipeline

## Motivation/summary

Update package `model` to no longer set `transaction.duration.us` or `span.duration.us`, and instead set `event.duration`. Update our ingest pipelines to use `event.duration` to set the legacy fields, and then remove `event.duration`. This pipeline can be removed once when we are ready and able to update the APM UI to use `event.duration` directly.

I have listed this as a breaking change in the top-level changelog, as it will have visible effects for anyone not using the Elasticsearch output or the packaged ingest pipeline. I have not done this in the integration package changelog, as I assume the ingest pipeline is always in use there.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

Send transactions and spans with zero and non-zero durations, and ensure they each have `transaction.duration.us` or `span.duration.us` set, and do not have `event.duration` set.

## Related issues

Prerequisite of https://github.com/elastic/apm-server/issues/5999
Part of https://github.com/elastic/apm-server/issues/3565 and https://github.com/elastic/apm-server/issues/4120
